setup.py

```diff
@@ -50,13 +50,18 @@
     )
     sys.exit(1)
 
+def set_builtin(name, value):
+    if isinstance(__builtins__, dict):
+        __builtins__[name] = value
+    else:
+        setattr(__builtins__, name, value)
 
 class build_ext(_build_ext):
     def finalize_options(self):
         print("Finalizing options")
         _build_ext.finalize_options(self)
         # Prevent numpy from thinking it is still in its setup process:
-        __builtins__.__NUMPY_SETUP__ = False
+        set_builtin('__NUMPY_SETUP__', False)
```

Comment:
Please use double quotes to keep the code `black` compliant.
```suggestion
        set_builtin("__NUMPY_SETUP__", False)
```